tests/Credentials/InstanceProfileProviderTest.php

```diff
@@ -138,6 +145,8 @@ private function getSecureTestClient(
                 switch ($request->getUri()->getPath()) {
                     case '/latest/meta-data/iam/security-credentials':
                     case '/latest/meta-data/iam/security-credentials/':
+                    case '/latest/meta-data/iam/security-credentials-extended':
+                    case '/latest/meta-data/iam/security-credentials-extended/':
```

Comment:
nit: Not a big deal, but I think would be easier if you just check whether `$request->getUri()->getPath()` has a forward slash, and if so then you just opt-in for either cases where the path will end with the forward slash or without it. For example if opting for leaving the path at the end:
```php
if (!str_ends_with(($path = $request->getUri()->getPath()), '/')) {
    $path = $path . "/";
}

switch ($path) {
    case "/latest/meta-data/iam/security-credentials/":
        break;
    case "/latest/meta-data/iam/security-credentials-extended/":
        break;
}
```